The error I see in the CI is:
```
tests/fft_test.py:201: in <lambda>
    lambda: func(rng([2, 3], dtype=np.float64), axis=[1, 1])
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

    @_wraps(np.fft.hfft)
    def hfft(a, n=None, axis=-1, norm=None):
      conj_a = lax.conj(a)
>     nn = (a.shape[axis] - 1) * 2 if n is None else n
E     TypeError: tuple indices must be integers or slices, not list
```
which is an error in your PR: it appears the test is passing `axis=[1, 1]`, but the implementation of `hfft` does not handle that case.

Note the test case is explicitly testing handling of illegal axis specifications, so the correct thing for your code to do here is to raise an exception if it sees an axis that isn't an integer (which is all `hfft` seems to accept: https://numpy.org/doc/stable/reference/generated/numpy.fft.hfft.html#numpy.fft.hfft ), and modify or extend the tests to expect that error from `hfft`.

--- a/docs/jax.numpy.rst
+++ b/docs/jax.numpy.rst
@@ -312,6 +312,8 @@ jax.numpy.fft
   irfft2
   rfftn
   irfftn
+  hfft
+  ihfft
   fftfreq
   rfftfreq
   fftshift

--- a/jax/numpy/fft.py
+++ b/jax/numpy/fft.py
@@ -131,14 +131,14 @@ def hfft(a, n=None, axis=-1, norm=None):
   conj_a = jnp.conj(a)
   _axis_check_1d('hfft', axis)
   nn = (a.shape[axis] - 1) * 2 if n is None else n
-  return _fft_core_1d('irfft', xla_client.FftType.IRFFT, conj_a, s=n, axis=axis,
+  return _fft_core_1d('hfft', xla_client.FftType.IRFFT, conj_a, s=n, axis=axis,
                       norm=norm) * nn
 
 @_wraps(np.fft.ihfft)
 def ihfft(a, n=None, axis=-1, norm=None):
   _axis_check_1d('ihfft', axis)
   nn = a.shape[axis] if n is None else n
-  output = _fft_core_1d('rfft', xla_client.FftType.RFFT, a, s=n, axis=axis,
+  output = _fft_core_1d('ihfft', xla_client.FftType.RFFT, a, s=n, axis=axis,
                       norm=norm)
   return jnp.conj(output) * (1 / nn)
 

--- a/jax/numpy/fft.py
+++ b/jax/numpy/fft.py
@@ -92,13 +92,16 @@ def irfftn(a, s=None, axes=None, norm=None):
   return _fft_core('irfftn', xla_client.FftType.IRFFT, a, s, axes, norm)
 
 
-def _fft_core_1d(func_name, fft_type, a, s, axis, norm):
+def _axis_check_1d(func_name, axis):
   full_name = "jax.numpy.fft." + func_name
   if isinstance(axis, (list, tuple)):
     raise ValueError(
         "%s does not support multiple axes. Please use %sn. "
         "Got axis = %r." % (full_name, full_name, axis)
     )
+
+def _fft_core_1d(func_name, fft_type, a, s, axis, norm):
+  _axis_check_1d(func_name, axis)
   axes = None if axis is None else [axis]
   return _fft_core(func_name, fft_type, a, s, axes, norm)
 
@@ -125,17 +128,19 @@ def irfft(a, n=None, axis=-1, norm=None):
 
 @_wraps(np.fft.hfft)
 def hfft(a, n=None, axis=-1, norm=None):
-  conj_a = lax.conj(a)
+  conj_a = jnp.conj(a)
+  _axis_check_1d('hfft', axis)
   nn = (a.shape[axis] - 1) * 2 if n is None else n
   return _fft_core_1d('irfft', xla_client.FftType.IRFFT, conj_a, s=n, axis=axis,
                       norm=norm) * nn
 
 @_wraps(np.fft.ihfft)
 def ihfft(a, n=None, axis=-1, norm=None):
+  _axis_check_1d('ihfft', axis)
   nn = a.shape[axis] if n is None else n
   output = _fft_core_1d('rfft', xla_client.FftType.RFFT, a, s=n, axis=axis,
                       norm=norm)
-  return lax.conj(output) * (1 / nn)
+  return jnp.conj(output) * (1 / nn)
 
 
 def _fft_core_2d(func_name, fft_type, a, s, axes, norm):

